lib/msf/ui/console/command_dispatcher/jobs.rb

```diff
@@ -257,7 +257,7 @@ def add_persist_job(job_id)
 
               payload_opts = {
                 'Payload'        => payload.refname,
-                'Options'        => payload.datastore,
+                'Options'        => payload.datastore.to_h,
```

Comment:
Serializing `payload.datastore` as JSON led to a string being saved, this allows a hash to be persisted instead